pkg/providers/translation/service.go

```diff
@@ -35,9 +35,15 @@ import (
 func (t *translator) TranslateService(namespace, name, subset string, port int32) (*apisixv1.Upstream, error) {
 	endpoint, err := t.EndpointLister.GetEndpoint(namespace, name)
 	if err != nil {
-		return nil, &TranslateError{
-			Field:  "endpoints",
-			Reason: err.Error(),
+		if k8serrors.IsNotFound(err) {
+			//Do not error wrap so that IsNotFound error can be handled separately on upper level
+			//Reference https://github.com/apache/apisix-ingress-controller/issues/1625
+			return nil, err
+		} else {
```

Comment:
It seems that this `else` is not necessary. Can you remove it?

Reply:
But if the error is `NotFound`, this function will return immediately. Otherwise this function will return a `&TranslateEror`. right?